test/Makefile

```diff
@@ -2,6 +2,13 @@ provision = true
 # If you set provision to false the test will run without compile the code
 # again.
 
+all: build
```

Comment:
I'm on the fence here, does this means when I run `make all` on the root directory it will also build ginkgo? If yes can we move the `build` target to a `test` target?

Reply:
Yes, that's the intention. It would build the tests, but not run ginkgo. 

There's a couple of instances that I think motivate this:
* Late last year there was a case where someone refactored code in `pkg/` and ended up unintentionally breaking the building of tests. 
* Just this week I've done the same thing with PRs, eg renaming a variable and missing some cases.

I think that the easiest way to pick up on stuff like this is for developers to run make on their local machines before pushing, and for the default make command to also at least compile the tests.

The main argument I see against this is that it makes the build take longer. At this point it's a couple of seconds.

Reply:
@aanm What would you like the behaviour to be for make all / make -C test all ?

Reply:
IMO `make all` would be to generate the binaries, `make tests` would execute the tests (including all the necessary ginkgo building). Gingko itself is not required to run cilium, that's why I think it doesn't make senses to build ginkgo as well

Reply:
OK, the latest version should achieve this. Specifically:

* `make` does not build the ginkgo tests.
* `make tests` builds ginkgo tests, but does not run them.
* `make -C test` / `cd test/ && make` will build ginkgo tests but not run them.
* `make -C test/ test` / `make -C test/ run` / `make -C test/ k8s` builds & runs ginkgo tests.

Reply:
@aanm Do you agree with this?

Reply:
yes I think it looks good, thanks!